openverse_catalog/dags/templates/template_provider.py_template

```diff
@@ -0,0 +1,148 @@
+        # TODO: Important! Refer to the most up-to-date documentation about the
+        # available fields in `openverse_catalog/docs/data_models.md`
+
+        # REQUIRED FIELDS:
+        # - foreign_landing_url
+        # - license_info
+        # - image_url / audio_url
+        #
+        # If a required field is missing, return early to prevent unnecesary
+        # processing.
+
+        if (foreign_landing_url := data.get("url")) is None:
+            return None
+
+        # TODO: Note the url field name differs depending on field type. Append
+        # `image_url` or `audio_url` depending on the type of record being processed.
+        if (image_url := data.get("image_url")) is None:
+            return None
+
+        # Use the `get_license_info` utility to get license information from a URL.
+        license_url = data.get("license")
+        license_info = get_license_info(license_url)
+        if license_info is None:
+            return None
+
+        # OPTIONAL FIELDS
+        # Obtain as many optional fields as possible.
+        foreign_identifier = data.get("foreign_id")
+        thumbnail_url = data.get("thumbnail")
+        filesize = data.get("filesize")
+        filetype = data.get("filetype")
+        creator = data.get("creator")
+        creator_url = data.get("creator_url")
+        title = data.get("title")
+        meta_data = data.get("meta_data")
+        raw_tags = data.get("tags")
+        watermarked = data.get("watermarked")
+
+        # MEDIA TYPE-SPECIFIC FIELDS
+        # Each Media type may also have its own optional fields. See documentation.
+        # TODO: Populate media type-specific fields.
+        # If your provider supports more than one media type, you'll need to first
+        # determine the media type of the record being processed.
+        #
+        # Example:
+        # media_type = self.get_media_type(data)
+        # media_type_specific_fields = self.get_media_specific_fields(media_type, data)
+        #
+        # If only one media type is supported, simply extract the fields here.
+
+        return {
+            "foreign_landing_url": foreign_landing_url,
+            "image_url": image_url,
+            "license_info": license_info,
+            # Optional fields
+            "foreign_identifier": foreign_identifier,
+            "thumbnail_url": thumbnail_url,
+            "filesize": filesize,
+            "filetype": filetype,
+            "creator": creator,
+            "creator_url": creator_url,
+            "title": title,
+            "meta_data": meta_data,
+            "raw_tags": raw_tags,
+            "watermarked": watermarked,
+            # TODO: Remember to add any media-type specific fields here
+        }
+
+
+def main():
+    # Allows running ingestion from the CLI without Airflow running for debugging
+    # purposes.
+    logger.info("Begin: {provider} data ingestion")
+    ingester = {provider}DataIngester()
+    ingester.ingest_records()
```

Comment:
Is there a `just` recipe we could add to make this easier? As a non-python-person I don't follow how to actually try and run this without airflow.